go/parquet/schema/column.go

```diff
@@ -0,0 +1,125 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/apache/arrow/go/parquet"
+	format "github.com/apache/arrow/go/parquet/internal/gen-go/parquet"
+)
+
+// Column encapsulates the information necessary to interpret primitive
+// column data in the context of a particular schema. We have to examine
+// the node structure of a column's path to the root in the schema tree
+// to be able to reassemble the nested structure from the repetition and
+// definition levels.
+type Column struct {
+	pnode     *PrimitiveNode
+	maxDefLvl int16
```

Comment:
useful to comment maxDevLvl and maxRepLvl I might be misremembering but I think we might have renamed these simply to defLvl and RepLvl in the C++ code.

Reply:
sure i'll add those comments.

Reply:
currently in the c++ code on master at least, they are still called `max_definition_level` and `max_repetition_level` https://github.com/apache/arrow/blob/master/cpp/src/parquet/schema.h#L406 which makes sense. They are the *max* definition and repetition level that exists in the column, since the individual levels would be on a per value basis.

Reply:
OK.  might be worth adding, that at this level, a corresponding value is present.

Reply:
comment added.